SCClassLibrary/Common/Math/Rational.sc

```diff
@@ -0,0 +1,194 @@
+Rational : Number {
+	var <>numerator, <>denominator;
+
+	*new { arg numerator, denominator = 1;
+		^super.new.initRational(numerator, denominator);
+	}
+
+	*newFromString { arg thisString;
+		^thisString.asRational;
+	}
+
+	initRational { arg thisNumerator, thisDenominator;
+		var factor;
+		if(thisNumerator.isString) {
+			^thisNumerator.asRational
+		} {
+			if(thisNumerator.isKindOf(Rational) or: thisDenominator.isKindOf(Rational)) {
+				^(thisNumerator.asRational / thisDenominator.asRational)
+			} {
+				if(thisNumerator.frac == 0 and: thisDenominator.frac == 0) {
+					if(thisDenominator == 0) {
+						this.numerator_(0);
+						this.denominator_(0);
+					} {
+						factor = gcd(thisNumerator.abs.asInteger, thisDenominator.abs.asInteger);
+						(thisDenominator < 0).if({factor = factor.neg});
+						(thisNumerator   < 0).if({factor = factor.neg});
+						this.numerator_((((thisNumerator/factor).abs) * factor.sign).asInteger);
+						this.denominator_((thisDenominator/factor).abs.asInteger);
+					};
+				} {
+					^(thisNumerator / thisDenominator).asRational
+				};
+			};
+		};
+	}
+	
+	performBinaryOpOnSimpleNumber { arg aSelector, aNumber, adverb;
+		^aNumber.asRational.perform(aSelector, this, adverb)
+	}
+	
+	isRational { ^true }
+	
+	%/ { arg aNumber; ^Rational(this, aNumber) }
+	
+	asRational { ^this }
+	
+	as { arg aSimilarClass; ^aSimilarClass.new(this.numerator, this.denominator) }
+	
+	asArray { ^[this.numerator, this.denominator]}
+	
+	asFloat { ^(this.numerator / this.denominator)}
```

Comment:
no need for the extra parentheses. The `^` (caret) is always on the left anyway.